### ISSUE TITLE
AutoMigrate will not update changed `default` value for primary keys

### ISSUE THREAD

**Issue by deefdragon** [go-gorm/gorm#6963]
## GORM Playground Link

(I was unable to get playground to function, trying to run ./test.sh caused a package not-found error in the build. I apologize. Ive tried to describe the issue in as depth as I can otherwise)

## Description

I recently changed all of my tables to use a default for the UUID primary key so that if I forget to set a value, the default takes over and generates a UUID for me. Specifically, this is for the primary keys (I have tested my example with non primary keys and changing/adding a default works fine). I have tested this example with postgres 15.

```go
//create a struct such as this (Notice the UUID primary key)
type Example struct {
	CreatedAt *time.Time      `json:"-"`
	DeletedAt *gorm.DeletedAt `gorm:"index" json:"-"`
	UpdatedAt *time.Time      `json:"UpdatedAt,omitempty,format:unix"`
	UUID      uuid.UUID       `gorm:"primarykey;type:uuid" json:"UUID"`
	Data string
}

//migrate the struct
lc.Gorm.AutoMigrate(&Example{})

//Update Example struct to the following (notice the addition of the default to UUID)
type Example struct {
	CreatedAt *time.Time      `json:"-"`
	DeletedAt *gorm.DeletedAt `gorm:"index" json:"-"`
	UpdatedAt *time.Time      `json:"UpdatedAt,omitempty,format:unix"`
	UUID uuid.UUID `gorm:"primarykey;type:uuid;default:gen_random_uuid()" json:"UUID"`
	Data string
}

lc.Gorm.AutoMigrate(&Example{})
``` 

After checking the database table data (via pgadmin), the `default` of UUID has not been changed, and is still empty.

From my perusal of the migrator code, this lack of change appears to be the case due to the `!field.PrimaryKey` check [here](https://github.com/go-gorm/gorm/blob/master/migrator/migrator.go#L513), which makes it appear intentional, but I was not able to find any documentation of this limitation, or reasoning as to why this limit exists. Its even stranger as you _can_ set the default for a primary key on table creation.

Is it possible this check was made to prevent setting `default:someConst` for the primary key?

**Comment by a631807682** [comment]
Yes, this is intentional, migrating primary keys is a dangerous practice. 
We should describe it clearly in the [documentation](https://gorm.io/docs/migration.html#Auto-Migration)

**Comment by a631807682** [comment]
You can use `AlterColumn` to change it.
Migrating the primary key is a relatively complex and dangerous operation. It may be a foreign key to another table, which may require confirmation of the order of table changes and transactions involved. Even so, there may still be other tables that are not declared in the same process.
Usually in the early stages of development we can recreate it with `DropTable` and `AutoMigrate`. Your suggestion may be possible, but we may need to confirm that this will bring enough benefits.

**Comment by MonstraG** [comment]
I just hit this problem, please add it to the docs. Or maybe fail in AutoMigration if this kind of change is required?

**Comment by krishna102001** [comment]
I was also facing this issue. Add this to documentation
